packages/program/src/revert/revert-error.ts

```diff
@@ -62,8 +62,18 @@ export class RevertError extends Error {
    * @param receipt - The transaction revert receipt.
    * @param reason - The revert reason.
    */
-  constructor(receipt: TransactionResultRevertReceipt, reason: RevertReason) {
-    super(`The script reverted with reason ${reason}`);
+  constructor(receipt: TransactionResultRevertReceipt, reason: RevertReason, logs: Array<unknown>) {
+    // @TODO: this is a "hack"
+    // motivation: a custom error revert string is the only "string" type item in the `logs` array
+    // the `logs.filter for string` will make sense only for `RequireFailed` errors, so could potentially
+    // just restrict the following to `RequireFailed` class by modifying `this.message`
+    // produces error messages like so:
+    // ```RequireRevertError: The script reverted with reason RequireFailed (reason: "FailImplicitly")```
+    super(
+      `The script reverted with reason ${reason}. (Reason: "${logs.filter(
+        (l) => typeof l === 'string'
+      )}")`
```

Comment:
```suggestion
      `The script reverted with reason ${reason}. (Reason: "${this.extractErrorFromLogs(logs)}")`
```

Might be easier to read for the end consumer

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1840/commits/3ae162dba97a55a08c7f7e3fb521f0b639b2be55